code/modules/ruins/spaceruin_code/hilbertshotel.dm

```diff
@@ -1,5 +1,6 @@
 GLOBAL_VAR_INIT(hhStorageTurf, null)
 GLOBAL_VAR_INIT(hhmysteryRoomNumber, 1337)
+#define MAX_ROOM_NUMBER 65535
```

Comment:
You can use up to `2^24 - 1` (`16777215`) without losing accuracy, according to Lummox.
http://www.byond.com/forum/?post=1877427